pyproject.toml

```diff
@@ -207,14 +207,8 @@ dev = [
   # Code formatting
   "black[jupyter]==22.6.0",
   # Documentation
-  "pydoc-markdown==4.5.1",  # FIXME Unpin!
-  # azure-core is a dependency of azure-ai-formrecognizer
-  # In order to stop malicious pip backtracking during pip install farm-haystack[all] documented in https://github.com/deepset-ai/haystack/issues/2280
-  # we have to resolve a dependency version conflict ourself.
-  # azure-core>=1.23 conflicts with pydoc-markdown's dependency on databind>=1.5.0 which itself requires typing-extensions<4.0.0
-  # azure-core>=1.23 needs typing-extensions>=4.0.1
-  # pip unfortunately backtracks into the databind direction ultimately getting lost.
-  "azure-core<1.23",
+  "pydoc-markdown",
+  "azure-core",
```

Comment:
If this pin is not needed, then let's remove `azure-core` from this group altogether. It's a transitive dependency.

Reply:
Hi @ZanSara! Indeed, this dependency declaration was just needed because of the dependency chain, the new changes made it non-necessary.